### PR TITLE
 Add port to support the server doesn't using 80

### PR DESCRIPTION
**Pull request by ZhihuiTian**

### CODE REVIEW

**Comment by phpmaple** [comment]
approved?

File: lib/sentry.dart [hunk 1]
```diff
@@ -161,7 +161,7 @@ class SentryClient {
 
   @visibleForTesting
   String get postUri =>
-      '${dsnUri.scheme}://${dsnUri.host}/api/$projectId/store/';
+      '${dsnUri.scheme}://${dsnUri.host}${dsnUri.port != 80 && dsnUri.port != 443 ? ':' + dsnUri.port.toString() : ''}/api/$projectId/store/';
```
Comment (yjbanov) [inline lib/sentry.dart:164]: This is probably a good time to switch from the `=>` shorthand to a `{ }` block syntax, and make port a variable.

Comment (yjbanov) [inline lib/sentry.dart:164]: `':' + dsnUri.port.toString()` can be replaced with `':${dsnUri.port}'`

**Review by yjbanov — CHANGES_REQUESTED** [review]
Thank you for the PR and sorry for the delay. This looks like a good feature.

Can you please add tests for this?

**Comment by MrMoronIV** [comment]
Can this be pulled into master? 

My SDN: 
```
https://dc31d4db460d4c198241b52537d90bc6:a08b9b9454e64dbea2a349d95cb5ba59@err.domain.com:9500/6
```
After a while I get:
```
Unhandled Exception: SocketException: OS Error: Connection refused, errno = 111, address = err.domain.com, port = 50563
```
I tried again:
```
Unhandled Exception: SocketException: OS Error: Connection refused, errno = 111, address = err.domain.com, port = 50564
```

without defining a port it says: `port = 50566`

**Comment by yjbanov** [comment]
It seems https://github.com/flutter/sentry/pull/42 covers this feature, as well as adds support for custom paths. So I'll close this PR in favor of https://github.com/flutter/sentry/pull/42.

**Pull request closed without merging**